### PR TITLE
vim-style +args

### PR DESCRIPTION
**Pull request by BenLubar**
For example, `./dfhack +echo foo` would be equivalent to running `./dfhack` and then typing `echo foo` into the console. uses shell argument splitting, so `./dfhack +somecommand "foo bar" baz` does the right thing.

See DFHack#755.

### CODE REVIEW

**Comment by expwnent** [comment]
What's wrong with putting it in dfhack.init?

**Comment by BenLubar** [comment]
It's easier to run a command once by putting it on the command line than it is to edit dfhack.init and then start the game and edit dfhack.init again. vim has .vimrc and +foo for the same reason.

The other part of #755 is that a plugin would have to define a `load region1` command, and then you could run `./dfhack +load region1` to skip the entire menu system and go straight to the loading screen for the world. Something like that would generally be a one-time use, not something you'd want to run every time you started the game.

File: library/Core.cpp [hunk 1]
```diff
@@ -1549,6 +1550,39 @@ bool Core::Init()
     if (!server->listen(RemoteClient::GetDefaultPort()))
         cerr << "TCP listen failed.\n";
 
+    vector<string *> & args = df::global::ui_sidebar_menus->unk.anon_2;
```
Comment (lethosor) [inline library/Core.cpp]: This should check for the goal being non-null (and the fields should probably be renamed too).

**Comment by BenLubar** [comment]
Ok, so DF does some weird stuff with command line arguments.

My new code should handle everything, but the way DF parses `./dfhack +foo "-bar -baz"` means that `foo` gets two arguments, each starting with a hyphen. There's no way to differentiate between that and `./dfhack +foo -bar -baz`, unfortunately.

However, I don't think there are any dfhack commands that need ` -` inside any of their arguments, so this shouldn't be a problem.

**Pull request merged**